python/cudf/cudf/core/udf/groupby_typing.py

```diff
@@ -0,0 +1,227 @@
+# Copyright (c) 2020-2023, NVIDIA CORPORATION.
+from typing import Any, Dict
+
+import numba
+import pandas as pd
+from numba import cuda, types
+from numba.core.extending import (
+    make_attribute_wrapper,
+    models,
+    register_model,
+    type_callable,
+    typeof_impl,
+)
+from numba.core.typing import signature as nb_signature
+from numba.core.typing.templates import AbstractTemplate, AttributeTemplate
+from numba.cuda.cudadecl import registry as cuda_registry
+from numba.np import numpy_support
+
+index_default_type = numpy_support.from_dtype(
+    pd.RangeIndex(0, 0).dtype
+)  # int64
+SUPPORTED_GROUPBY_NUMBA_TYPES = [types.int64, types.float64]
+SUPPORTED_GROUPBY_NUMPY_TYPES = [
+    numpy_support.as_dtype(dt) for dt in [types.int64, types.float64]
+]
+
+
+class Group(object):
```

Comment:
Minor nit: should we consider a more specific name here than `Group`? I don't think it's a big deal since it's so heavily nested in the package, but it could get a bit confusing if code does `from ... import Group`.